app/routes.py

```diff
@@ -1,2 +1,338 @@
+from datetime import datetime
+import requests
+# from requests import post
+import os
 
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix=("/tasks"))
+goals_bp = Blueprint("goals_bp", __name__, url_prefix=("/goals"))
+
+@tasks_bp.route("", methods=["GET", "POST"])
+#any time we make a request to our api and call localhost5000, we are going to run the function.
+#what will run depends on if the client calls GET or POST
+def handle_tasks(): #view function names should be unique
+    if request.method == "GET":
+        tasks = Task.query.all() #Task is a child of SQLA so we can use the query method, 
+        #query is method from sqla, paired with .all, retrieves all rows from task table in database, from there SQLA will turn all those rows into instances
+        #of the task class, and all method will return all instances of the Task class into a list, and that list is assigned to variable task
+        #Task.query.all.sort?? we can order the result the result of our selection
+        tasks_response = []
+        for task in tasks: #<Task 2342> - returns object, if we want to see attributes we have to 
+            #access the attributes. So we break down objhects in that list in dictionary so we can see each instance's attributes
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            })
+        
+        sort_param = request.args.get("sort")
+        if sort_param == "asc":
+            tasks_response.sort(key=lambda t: t["title"])
+            
+        if sort_param == "desc":
+            tasks_response.sort(key=lambda t: t["title"], reverse=True)
+
+        return jsonify(tasks_response), 200 #describes respond of an endpoint, once we receive this request, what do we respond with
+#take data we want to respond with and put it in a javascript object format aka JSON, turns data into JSON form b/c JSON is an accepted data type to communicate between two entities
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            # return make_response("400 Bad Request")
+            return {"details": "Invalid data"}, 400
+        
+        new_task = Task(
+            title= request_body["title"],
+            description= request_body["description"],
+            completed_at= request_body["completed_at"]
+        )
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {
+            "task": {
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": False if new_task.completed_at == None else True
+            }
+        }, 201
```

Comment:
We can reduce this return statement to 1 line by using a helper method in the Task class:

```suggestion
        return {"task": task.to_dict()}, 201
```